### PR TITLE
add network stop on server kill signal

### PR DESCRIPTION
**Pull request by felipemadero**

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -177,6 +178,14 @@ func (s *server) Run(rootCtx context.Context) (err error) {
 
 		s.gRPCServer.Stop()
 		zap.L().Warn("closed gRPC server")
+
+		if s.network != nil {
+			stopCtx, stopCtxCancel := context.WithTimeout(context.Background(), StopOnSignalTimeout)
+			defer stopCtxCancel()
+			s.network.stop(stopCtx)
+			zap.L().Warn("stoping network")
```
Comment (gyuho) [inline server/server.go:186]: Is `stop` async operation? Should we add warning before we call `stop`?

Comment (holisticode) [inline server/server.go]: I think it should say "stopping"

Comment (holisticode) [reply server/server.go]: I think it's not async from here.

Comment (felipemadero) [reply server/server.go]: stop is blocking. changed msg to "network stopped"

File: server/server.go [hunk 2]
```diff
@@ -177,6 +178,14 @@ func (s *server) Run(rootCtx context.Context) (err error) {
 
 		s.gRPCServer.Stop()
 		zap.L().Warn("closed gRPC server")
+
+		if s.network != nil {
```
Comment (holisticode) [inline server/server.go:182]: Question: why are we applying this only to the `case err = <-gwErrc` case? Shouldn't this be done in all cases?

File: server/server.go (hunk 1 above)
Comment (felipemadero) [reply server/server.go]: already changed

File: server/server.go (hunk 2 above)
Comment (felipemadero) [reply server/server.go]: we were only applying only to the rootCtx closure (dure to signal handling). But you are right. Moving this to outside the select.

**Review by holisticode — APPROVED** [review]

**Pull request merged**